plugins/aws/cloudfront/cloudfrontHttpsOnly.js

```diff
@@ -52,5 +71,64 @@ module.exports = {
         });
 
         callback(null, results, source);
+    },
+    remediate: function(config, cache, settings, resource, callback) {
+        var putCall = this.actions.remediate;
+        var pluginName = 'cloudfrontHttpsOnly';
+        var cdnNameArr = resource.split(':');
+        var cdnName = cdnNameArr[cdnNameArr.length - 1].split('/');
+        var cdnId = cdnName[1];
+        var cdnLocation = helpers.defaultRegion(settings);
+
+        var getDistribution = helpers.addSource(cache, {},
+            ['cloudfront', 'getDistribution', cdnLocation, cdnId]);
+
+        var err;
+        var params = {};
+
+        if (getDistribution &&
+            getDistribution.data &&
+            getDistribution.data.ETag &&
+            getDistribution.data.Distribution &&
+            getDistribution.data.Distribution.DistributionConfig) {
+            params['DistributionConfig'] = getDistribution.data.Distribution.DistributionConfig;
+            
+            if (settings.input && settings.input.cdnPolicyOption) params['DistributionConfig']['ViewerProtocolPolicy'] = settings.input.cdnPolicyOption;
+            else params['DistributionConfig']['DefaultCacheBehavior']['ViewerProtocolPolicy'] = 'redirect-to-https';
+
+            params['DistributionConfig']['Comment'] = 'AquaCloud Remediation';
+            params['Id'] = cdnId;
+            params['IfMatch'] = getDistribution.data.ETag;
```

Comment:
Is IfMatch this a mandatory field? if not can we just leave it. I think for remediation we are only feeling the mandatory params.